File: Java/src/test/java/org/chornogora/task_1_two_sum/SolutionTest.java
package org.chornogora.task_1_two_sum;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class SolutionTest {

    @Test
    public void shouldReturn_0_1() {
        Solution solution = new Solution();
        int[] result = solution.twoSum(new int[]{11, 2, 7, 15}, 9);
        Assertions.assertEquals(2, result.length);
        Assertions.assertEquals(1, result[0]);
        Assertions.assertEquals(2, result[1]);
    }
}

File: Kotlin/src/test/kotlin/org/chornogora/task_3_longest_substring/SolutionTest.kt
package org.chornogora.task_3_longest_substring

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class SolutionTest {

    private val solution = Solution()

    @Test
    fun shouldReturnZeroWhenEmptyLineProvided() {
        val result = solution.lengthOfLongestSubstring("")
        assertEquals(0, result)
    }

    @Test
    fun shouldReturnThree() {
        val result = solution.lengthOfLongestSubstring("abcabcbb")
        assertEquals(3, result)
    }

    @Test
    fun shouldReturnThreeAsWell() {
        val result = solution.lengthOfLongestSubstring("pwwkew")
        assertEquals(3, result)
    }

    @Test
    fun shouldReturnThreeToo() {
        val result = solution.lengthOfLongestSubstring("dvdf")
        assertEquals(3, result)
    }

    @Test
    fun shouldReturnTwo() {
        val result = solution.lengthOfLongestSubstring("abba")
        assertEquals(2, result)
    }

    @Test
    fun shouldReturnSeven() {
        val result = solution.lengthOfLongestSubstring("vbxpvwkkteaiob")
        assertEquals(7, result)
    }

    @Test
    fun shouldReturnOneForSimilarSymbols() {
        val result = solution.lengthOfLongestSubstring("bbbbb")
        assertEquals(1, result)
    }
}